libs/common/files.cpp

```diff
@@ -40,3 +47,16 @@ void iroha::remove_dir_contents(const std::string &dir,
       log->error("{}", error_code.message());
   }
 }
+
+iroha::expected::Result<const std::string &, const std::string &>
+iroha::readFile(const std::string &path) {
+  std::ifstream file(path);
+  if (!file.good()) {
+    return iroha::expected::makeError(
+        (boost::format(kCannotReadFile) % path).str());
+  }
+
+  std::stringstream ss;
+  ss << file.rdbuf();
```

Comment:
Isn't this redundant with the check of the `file` object above?

```cpp
if (!file) {
```

https://en.cppreference.com/w/cpp/io/basic_ios/operator_bool - looks like if `file` evaluates to `true`, then it could either be read and produce a non-empty valid string here, or would result in an empty string (in case EOF is met)

Reply:
seems that such a connection between stream errors may exist, but though is not very evident. an explicit check of stream error does not add much resource overhead, neither complicates the code too much. my vote is for adding the check and staying with more evident security.